### PR TITLE
Rename .observe to .record

### LINKED ISSUE #3914: Rename ObservableMeasurement.observe -> ObservableMeasurement.record?
I feel as if `Consumer<Measurement>` is the observer and doing the observing - the method on the callback seems like it's actually recording, and is meant to be "semantically" the same as the record methods on the synchronous instruments.

### PR DESCRIPTION
**Pull request by anuraaga**
Also removed MUST be non-negative from docs, this is true only of Counter instruments I think.

Fixes #3914

### CODE REVIEW

**Review by jack-berg — APPROVED** [review]
The plan is to leave these deprecated for the release candidate, and remove them for the next published version (which will be the first version which includes metrics as part of the stable API)?

**Comment by jack-berg** [comment]
This should resolve #3914.

**Review by jkwatson — APPROVED** [review]

File: api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java [hunk 1]
```diff
@@ -12,15 +12,38 @@
...
   /**
    * Records a measurement with a set of attributes.
    *
-   * @param value The measurement amount. MUST be non-negative.
+   * @param value The measurement amount.
    * @param attributes A set of attributes to associate with the count.
+   * @deprecated Use {@link #record(double, Attributes)}.
    */
-  void observe(double value, Attributes attributes);
+  @Deprecated
```
Comment (breedx-splk) [inline api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java:30]: should we include a message about what to use instead?

File: api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java [hunk 2]
```diff
@@ -12,15 +12,38 @@
   /**
    * Records a measurement.
    *
-   * @param value The measurement amount. MUST be non-negative.
+   * @param value The measurement amount.
+   * @deprecated Use {@link #record(double)}.
    */
-  void observe(double value);
+  @Deprecated
```
Comment (breedx-splk) [inline api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java:18]: Include a message about using `record(value)` instead?

**Review by breedx-splk — APPROVED** [review]
Looks ok to me, but I didn't verify the MUST/non-negative thing. Would be nice to have a note in the deprecation about using the new name. 🥂

File: api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java (hunk 2 above)
Comment (anuraaga) [reply api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java:18]: Yeah it's there in the javadoc, this is the annotation 😛

**Pull request merged**